dimod/views/quadratic.py

```diff
@@ -390,3 +395,90 @@ def set_linear(self, v: Variable, bias: Bias):
     @abc.abstractmethod
     def set_quadratic(self, u: Variable, v: Variable, bias: Bias):
         raise NotImplementedError
+
+    def iter_linear(self) -> Iterator[Tuple[Variable, Bias]]:
+        """Iterate over the variables and their biases."""
+        get = self.get_linear
+        for v in self.variables:
+            yield v, get(v)
```

Comment:
Just curious: what's the advantage of defining this `get` and using `.get_linear` over using `key, val in self.linear.items()` ?

Reply:
very minor performance boosts. This method is pretty low level so it makes sense to squeeze our the optimizations but not in general worth doing.
```python
In [1]: import dimod

In [2]: bqm = dimod.generators.gnp_random_bqm(500, 1, 'BINARY')

In [3]: def f():
   ...:     for k, v in bqm.linear.items(): pass
   ...: 

In [4]: def g():
   ...:     for v in bqm.variables:
   ...:         bqm.get_linear(v)
   ...: 

In [5]: def h():
   ...:     get = bqm.get_linear
   ...:     for v in bqm.variables:
   ...:         get(v)
   ...: 

In [6]: %timeit f()
204 µs ± 5.5 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [7]: %timeit g()
101 µs ± 717 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [8]: %timeit h()
84.9 µs ± 483 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

Reply:
perfect, thanks!